subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -580,9 +580,17 @@ static inline void sw_switch(uint8_t dir_curr, uint8_t dir_next,
 #endif /* CONFIG_BT_CTLR_PHY_CODED */
 	} else {
 		/* RX */
+
+#if defined(CONFIG_SOC_SERIES_BSIM_NRFXX)
+		delay = HAL_RADIO_NS2US_CEIL(
+			hal_radio_rx_ready_delay_ns_get(phy_next, flags_next)) +
+			4;
+#else
 		delay = HAL_RADIO_NS2US_CEIL(
 			hal_radio_rx_ready_delay_ns_get(phy_next, flags_next) -
-			hal_radio_tx_chain_delay_ns_get(phy_curr, flags_curr));
+			hal_radio_rx_chain_delay_ns_get(phy_curr, flags_curr)) +
+			4;
+#endif
```

Comment:
I do not really want to start another discussion about timing, but this particular change looks like a workaround.

tx_chain_delay should compensate for EVENT_END being triggered before end of PDU in the air (TX path delay from digital radio to antenna) which is 0.6us, that was correct. Once you changed to rx_chain_delay which is 9.4us, it reduces compensation too much and radio starts RX somewhere during preamble. It works on hw, because it can receive PDU with incomplete preamble. Apparently bsim cannot do this.

![image](https://user-images.githubusercontent.com/9692685/118472829-94903700-b709-11eb-915f-835ddadbdadf.png)

Reply:
> Apparently bsim cannot do this.

In bsim, the phy model only considers a packet in air as possibly received if the receiver is open during the first bit of the preamble (this is a simplification, and probably a bit tougher than what some radios may be able to accept, which seems to include yours :) )

Reply:
What I am trying to understand here is, TXEN starts with first bit transmitted from transmitter and RADIO_END is the last bit received on the receiver. I am sure there is time spent between the first bit and last bit of the last byte of the PDU (CRC last byte). This (first bit to last bit of the last byte) is the amount of time that the receiver will have to subtract when turning around to transmitted back in order  to meet the TIFS which is end of last bit received to the transmit of first bit back. This is my definition of rx_chain_delay used when transmitting back, which depends on the transmitted PHY.

> Once you changed to rx_chain_delay which is 9.4us, it reduces compensation too much and radio starts RX somewhere during preamble.

In the code above, RXEN happens 16 us (10 us not compensated plus 10 us delayed minus the 4 us) delayed; note the ready_delay that is reduced by 10 us (for 1M PHY). This should, in theory, fail access address from being received unless radio is receiving earlier than the ready delay.

Now I am myself confused.

In thoery, then it should be

```
		delay = HAL_RADIO_NS2US_CEIL(
			hal_radio_rx_ready_delay_ns_get(phy_next, flags_next) +
			hal_radio_rx_chain_delay_ns_get(phy_curr, flags_curr)) +
			4;
```

Radio PPI should be triggered at (40 + 9.4  + 4) before the preamble. i.e.  (150 - 54), 96 us after the Tx Radio End. But for some reason, triggering the PPI at (40 - 9.4 + 4) before the preamble, (150 - 35), 115 us after the Tx Radio End still passes the conformance timing tests!